programs/castle-vault/src/adapters/port.rs

```diff
@@ -71,34 +103,112 @@ pub struct PortAccounts<'info> {
 impl_has_vault!(PortAccounts<'_>);
 
 impl<'info> LendingMarket for PortAccounts<'info> {
+    fn verify_accounts(&self, program_id: &Pubkey) -> ProgramResult {
+        let port_additional_states_pda_key = Pubkey::create_program_address(
+            &[
+                self.vault.key().as_ref(),
+                b"port_additional_state".as_ref(),
+                &[self.vault.vault_port_additional_state_bump],
+            ],
+            program_id,
+        )?;
+        let port_stake_account_pda_key = Pubkey::create_program_address(
+            &[
+                self.vault.key().as_ref(),
+                b"port_stake".as_ref(),
+                &[self.port_additional_states.vault_port_stake_account_bump],
+            ],
+            program_id,
+        )?;
+        let port_reward_account_pda_key = Pubkey::create_program_address(
+            &[
+                self.vault.key().as_ref(),
+                b"port_reward".as_ref(),
+                &[self.port_additional_states.vault_port_reward_token_bump],
+            ],
+            program_id,
+        )?;
+        let port_obligation_pda_key = Pubkey::create_program_address(
+            &[
+                self.vault.key().as_ref(),
+                b"port_obligation".as_ref(),
+                &[self.port_additional_states.vault_port_obligation_bump],
+            ],
+            program_id,
+        )?;
+
+        if port_additional_states_pda_key != self.port_additional_states.key()
+            || port_stake_account_pda_key != self.vault_port_stake_account.key()
+            || port_reward_account_pda_key != self.vault_port_reward_token.key()
+            || port_obligation_pda_key != self.vault_port_obligation.key()
+            || self.port_additional_states.port_lp_token_account != self.port_lp_token_account.key()
```

Comment:
These checks can be done with `has_one` over `port_additional_state` in the struct

Reply:
All of these are just PDAs and we only store the bump key. How to apply `has_one` in this case?